### PR TITLE
[FIX] mail: revert the unlink in batch of the <mail.mail>

### PR DESCRIPTION
**Pull request by std-odoo**
Bug
===
The unlink of the <mail.mail> in the CRON is problematic because we accumulate a lot of records, and the CRON timeout.

Task-3179157

### CODE REVIEW

File: addons/mail/views/mail_mail_views.xml [hunk 1]
```diff
@@ -46,8 +46,6 @@
                                     <group string="Status">
                                         <field name="auto_delete"
                                             attrs="{'invisible': [('state', '!=', 'outgoing'), ('state', '!=', 'exception')]}"/>
-                                        <field name="to_delete"
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_mail_views.xml:49]: For stable, we could set invisible=1, and update the view in master, to lessen potential issues (does not cost much).

File: addons/mail/views/mail_mail_views.xml [hunk 2]
```diff
@@ -99,7 +97,6 @@
                     <field name="message_type" invisible="1"/>
                     <field name="state" widget="badge" decoration-muted="state in ('sent', 'cancel')"
                         decoration-info="state=='outgoing'" decoration-danger="state=='exception'"/>
-                    <field name="to_delete"/>
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_mail_views.xml:102]: Same here, would keep as invisible.

File: addons/project/models/project.py [hunk 3]
```diff
@@ -2236,6 +2236,7 @@ def _task_message_auto_subscribe_notify(self, users_per_task):
                     record_name=task.display_name,
                     email_layout_xmlid='mail.mail_notification_layout',
                     model_description=task_model_description,
+                    mail_auto_delete=False,
```
Comment (tde-banana-odoo) [inline addons/project/models/project.py:2239]: Why this change ?

File: addons/mail/models/mail_mail.py [hunk 4]
```diff
@@ -84,6 +84,7 @@ def default_get(self, fields):
     auto_delete = fields.Boolean(
         'Auto Delete',
         help="This option permanently removes any track of email after it's been sent, including from the Technical menu in the Settings, in order to preserve storage space of your Odoo database.")
+    # TODO: remove in master
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py:87]: Would say something like "Unused since v16, to remove in master."

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing ! Some comments for the revert.

Concerning commit message I think we could be a bit more verbose, because it does not really explain what are the issues we faced... someone external won't understand why we suddenly revert this feature.

And don't forget to rebase, some performance counters seems to fail :D .

File: addons/mail/views/mail_mail_views.xml (hunk 1 above)
Comment (std-odoo) [reply addons/mail/views/mail_mail_views.xml:49]: yes, good idea :)

File: addons/project/models/project.py (hunk 3 above)
Comment (std-odoo) [reply addons/project/models/project.py:2250]: Because of the test  `test_personal_stage_search`
Looks like `personal_stage_id` expect the mail to exists in the database (it was the case before, because the CRON didn't run yet)
See: https://runbot.odoo.com/runbot/build/23750491

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Hi :) 

I did the changes / updated the commit message :)

File: addons/project/models/project.py (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/project/models/project.py:2250]: Strange but ok, we won't solve that here :D

**Comment by tde-banana-odoo** [comment]
@robodoo r+
